casepro/pods/base.py

```diff
@@ -0,0 +1,46 @@
+from confmodel import fields, Config as ConfmodelConfig
+from django.apps import AppConfig
+from django.conf.urls import url
+from django.http import HttpResponse
+
+
+class PodConfig(ConfmodelConfig):
+    '''
+    This is the config that all pods should use as the base for their own
+    config.
+    '''
+    index = fields.ConfigInt(
+        "A unique identifier for the specific instance of this pod."
+        "Automatically determined and set in the pod registry.",
+        required=True)
+
+
+class Pod(object):
+    '''
+    The base class for all pod plugins.
+    '''
+    config_cls = PodConfig
+    name = 'Default pod name'
+    controller = None
+    directive = None
+
+    def __init__(self, config):
+        self.config = self.config_cls(config)
+
+    @property
+    def url_patterns(self):
+        return (
+            url(r'read/%d/$' % self.config.index, self.request_callback),
```

Comment:
Would the data pod do something like add its index to the request, which we can then route on? If the data pod knows its index, then is there much difference between it putting it in the URL or putting it in the query args? I was looking at this for contructing the urls (although maybe I'm understanding it incorrectly): https://github.com/praekelt/casepro/pull/34/files#diff-605d5fe78f6a6aac57fb0e52c1ecceb3R428